src/pages/App.js

```diff
@@ -1,17 +1,30 @@
 import React, { useEffect } from "react";
 import { auth, persist } from "../firebase";
 import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
+import { actionTypes } from "../reducer";
 import { useStateValue } from "../StateProvider";
 import Chat from "../components/Chat";
 import Footer from "../components/Footer";
 import Login from "./Login";
 import Sidebar from "../components/Sidebar";
 
 function App() {
-  const [{ user }] = useStateValue();
+  const [{ user }, dispatch] = useStateValue();
 
   useEffect(() => {
     auth.setPersistence(persist);
+
+    const listener = auth.onAuthStateChanged((authUser) => {
```

Comment:
Woah so simple, so by adding listener. It actually checking the user if it was logged in before page reloaded ?

is the auth.setPersistence(persist); really working ?

Thanks

Reply:
Yeah! Firebase I believe persists log in until the user explicitly logs out. I do believe auth.setPersistence is not doing anything here.